api_tests/base/test_sloan_study.py

```diff
@@ -0,0 +1,210 @@
+        assert f'no_{SLOAN_COI}' in tags
+
+        assert SLOAN_DATA not in resp.json['meta']['active_flags']
+        assert SLOAN_PREREG not in resp.json['meta']['active_flags']
+        assert SLOAN_COI not in resp.json['meta']['active_flags']
+
+        cookies = resp.headers.getall('Set-Cookie')
+
+        assert f' {SLOAN_DATA}=False; Path=/' in cookies
+        assert f' {SLOAN_PREREG}=False; Path=/' in cookies
+        assert f' {SLOAN_COI}=False; Path=/' in cookies
+
+    @pytest.mark.enable_quickfiles_creation
+    @mock.patch('api.base.views.Flag.is_active')
+    def test_sloan_study_control_unauth(self, mock_flag_is_active, app, user, preprint):
+        mock_flag_is_active.return_value = False
+        headers = {'Referer': preprint.absolute_url}
+        resp = app.get('/v2/', headers=headers)
+
+        assert SLOAN_DATA not in resp.json['meta']['active_flags']
+        assert SLOAN_PREREG not in resp.json['meta']['active_flags']
+        assert SLOAN_COI not in resp.json['meta']['active_flags']
+
+        cookies = resp.headers.getall('Set-Cookie')
+
+        assert f' {SLOAN_DATA}=False; Path=/' in cookies
+        assert f' {SLOAN_PREREG}=False; Path=/' in cookies
+        assert f' {SLOAN_COI}=False; Path=/' in cookies
+
+    @pytest.mark.parametrize('reffer_url', [
+        f'{DOMAIN}preprints',
+        f'{DOMAIN}preprints/',
+        f'{DOMAIN}preprints/foorxiv',
+        f'{DOMAIN}preprints/foorxiv/',
+        f'{DOMAIN}preprints/foorxiv/aguid',
+        f'{DOMAIN}preprints/foorxiv/aguid/',
+        f'{DOMAIN}preprints/not/a/valid/path/'
+    ])
+    @mock.patch('api.base.views.Flag.is_active')
+    def test_weird_domains(self, mock_flag_is_active, app, reffer_url):
+        mock_flag_is_active.return_value = True
+        headers = {'Referer': reffer_url}
+        resp = app.get('/v2/', headers=headers)
+        assert resp.status_code == 200
+
+        cookies = resp.headers.getall('Set-Cookie')
+
+        assert f' {SLOAN_DATA}=True; Path=/' in cookies
+        assert f' {SLOAN_PREREG}=True; Path=/' in cookies
+        assert f' {SLOAN_COI}=True; Path=/' in cookies
+
+    @pytest.mark.parametrize('reffer_url', [
+        DOMAIN,
+        f'{DOMAIN}not-preprints/',
+    ])
+    @mock.patch('api.base.views.Flag.is_active')
+    def test_too_weird_domains(self, mock_flag_is_active, app, reffer_url):
+        mock_flag_is_active.return_value = True
+        headers = {'Referer': reffer_url}
+        resp = app.get('/v2/', headers=headers)
+        resp.status_code == 200
+
+        cookies = resp.headers.getall('Set-Cookie')
+
+        assert f' {SLOAN_DATA}=True; Path=/' not in cookies
+        assert f' {SLOAN_PREREG}=True; Path=/' not in cookies
+        assert f' {SLOAN_COI}=True; Path=/' not in cookies
+
+    @pytest.mark.enable_quickfiles_creation
+    @mock.patch('api.base.views.Flag.is_active')
+    def test_provider_custom_domain(self, mock_flag_is_active, app, user, preprint):
```

Comment:
maybe update this test to check that it's matching the correct provider, too (i don't think this is a problem right now, but it might be an easy regression)